src/Snak/ExtendedPropertyValueSnak.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Wikibase\DataModel\Snak;
+
+use DataValues\DataValue;
+use InvalidArgumentException;
+use Wikibase\DataModel\Entity\PropertyId;
+
+/**
+ * PropertyValueSnak with with extra attached DataValues.
+ * Extra DataValues can be used to provide infomation, for example, normalized values.
+ * Calls to the getType method will indicate that this is a PropertyValueSnak
+ *
+ * @since 3.1
+ *
+ * @licence GNU GPL v2+
+ * @author Adam Shorland
+ */
+class ExtendedPropertyValueSnak extends PropertyValueSnak {
+
+	/**
+	 * @var DataValue[]
+	 */
+	private $extensionDataValues = array();
+
+	/**
+	 * @param PropertyId $propertyId
+	 * @param DataValue $dataValue
+	 * @param DataValue[] $extensionDataValues
+	 */
+	public function __construct(
+		PropertyId $propertyId,
+		DataValue $dataValue,
+		array $extensionDataValues
+	) {
+		parent::__construct( $propertyId, $dataValue );
+
+		foreach ( $extensionDataValues as $key => $extensionDataValue ) {
+			if ( !( $extensionDataValue instanceof DataValue ) || !is_string( $key ) ) {
+				throw new InvalidArgumentException(
+					'$extensionDataValues must be an array of DataValue objects with string keys'
+				);
+			}
+		}
+
+		$this->extensionDataValues = $extensionDataValues;
+	}
+
+	/**
+	 * @return DataValue[] with string keys
+	 */
+	public function getExtensionDataValues() {
```

Comment:
"Extension" is odd... How about "extra data values", "secondary data values", or "derived data values"?